<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OS_report_designClass</class>
 <widget class="QWidget" name="OS_report_designClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1113</width>
    <height>520</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OS_report_design</string>
  </property>
  <widget class="QTabWidget" name="tabWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>581</width>
     <height>501</height>
    </rect>
   </property>
   <property name="currentIndex">
    <number>1</number>
   </property>
   <widget class="QWidget" name="pro">
    <property name="font">
     <font>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <attribute name="title">
     <string>Process scheduling algorithm</string>
    </attribute>
    <widget class="QWidget" name="horizontalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>40</y>
       <width>461</width>
       <height>31</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Name of processes</string>
        </property>
       </widget>
      </item>
      <item alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Arrive_time</string>
        </property>
       </widget>
      </item>
      <item alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Time required</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>131</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Number of processes</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox1">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>10</y>
       <width>69</width>
       <height>22</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>3</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>4</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>5</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>6</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>7</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>8</string>
      </property>
     </item>
    </widget>
    <widget class="QPushButton" name="pushButton">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>60</y>
       <width>91</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>FCFS</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_2">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>120</y>
       <width>91</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>SJF</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_3">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>180</y>
       <width>91</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>FPF</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_4">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>240</y>
       <width>91</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>RR</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_5">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>300</y>
       <width>91</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>MFQ</string>
     </property>
    </widget>
    <widget class="QWidget" name="gridLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>441</width>
       <height>401</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="Box_Grid"/>
    </widget>
   </widget>
   <widget class="QWidget" name="nam">
    <attribute name="title">
     <string>Page replacement algorithm</string>
    </attribute>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>20</y>
       <width>91</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Generate_Page :</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_6">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>20</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>AUTO_Generate</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_7">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>20</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Enter_Manually</string>
     </property>
    </widget>
    <widget class="QWidget" name="gridLayoutWidget_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>441</width>
       <height>371</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout"/>
    </widget>
    <widget class="QPushButton" name="pushButton_8">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>120</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Optimal</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_9">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>180</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>FIFO</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_10">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>240</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>LRU</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_11">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>300</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Clock</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_12">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>360</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Improved Clock</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="Line" name="line">
   <property name="geometry">
    <rect>
     <x>590</x>
     <y>0</y>
     <width>20</width>
     <height>521</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="QTextBrowser" name="textBrowser">
   <property name="geometry">
    <rect>
     <x>610</x>
     <y>10</y>
     <width>491</width>
     <height>261</height>
    </rect>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="OS_report_design.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>comboBox1</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>OS_report_designClass</receiver>
   <slot>SelectProcessNumber(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>180</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>189</x>
     <y>519</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>OS_report_designClass</receiver>
   <slot>FCFS()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>541</x>
     <y>114</y>
    </hint>
    <hint type="destinationlabel">
     <x>545</x>
     <y>519</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>OS_report_designClass</receiver>
   <slot>SJF()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>535</x>
     <y>171</y>
    </hint>
    <hint type="destinationlabel">
     <x>524</x>
     <y>519</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>OS_report_designClass</receiver>
   <slot>FPF()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>504</x>
     <y>221</y>
    </hint>
    <hint type="destinationlabel">
     <x>519</x>
     <y>517</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>OS_report_designClass</receiver>
   <slot>RR()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>518</x>
     <y>290</y>
    </hint>
    <hint type="destinationlabel">
     <x>514</x>
     <y>519</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>OS_report_designClass</receiver>
   <slot>MFQ()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>495</x>
     <y>352</y>
    </hint>
    <hint type="destinationlabel">
     <x>484</x>
     <y>519</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>OS_report_designClass</receiver>
   <slot>Data_Generate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>235</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>185</x>
     <y>642</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_7</sender>
   <signal>clicked()</signal>
   <receiver>OS_report_designClass</receiver>
   <slot>Plot_Inputbox()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>348</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>354</x>
     <y>546</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_8</sender>
   <signal>clicked()</signal>
   <receiver>OS_report_designClass</receiver>
   <slot>Optimal()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>515</x>
     <y>165</y>
    </hint>
    <hint type="destinationlabel">
     <x>531</x>
     <y>549</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_9</sender>
   <signal>clicked()</signal>
   <receiver>OS_report_designClass</receiver>
   <slot>FIFO()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>489</x>
     <y>231</y>
    </hint>
    <hint type="destinationlabel">
     <x>490</x>
     <y>583</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_10</sender>
   <signal>clicked()</signal>
   <receiver>OS_report_designClass</receiver>
   <slot>LRU()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>503</x>
     <y>295</y>
    </hint>
    <hint type="destinationlabel">
     <x>500</x>
     <y>573</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_11</sender>
   <signal>clicked()</signal>
   <receiver>OS_report_designClass</receiver>
   <slot>Clock()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>547</x>
     <y>344</y>
    </hint>
    <hint type="destinationlabel">
     <x>535</x>
     <y>577</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_12</sender>
   <signal>clicked()</signal>
   <receiver>OS_report_designClass</receiver>
   <slot>Improved_Clock()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>555</x>
     <y>408</y>
    </hint>
    <hint type="destinationlabel">
     <x>557</x>
     <y>538</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>viewsig()</signal>
  <slot>SelectProcessNumber(int)</slot>
  <slot>FCFS()</slot>
  <slot>SJF()</slot>
  <slot>FPF()</slot>
  <slot>RR()</slot>
  <slot>MFQ()</slot>
  <slot>Data_Generate()</slot>
  <slot>Plot_Inputbox()</slot>
  <slot>Optimal()</slot>
  <slot>FIFO()</slot>
  <slot>LRU()</slot>
  <slot>Clock()</slot>
  <slot>Improved_Clock()</slot>
 </slots>
</ui>
